src/ui/Theme.h

```diff
@@ -23,12 +22,20 @@ class Theme final : public QPalette
     Q_PROPERTY(QColor online READ online CONSTANT)
     Q_PROPERTY(QColor unavailable READ unavailable CONSTANT)
 public:
-    Theme() {}
-    explicit Theme(QStringView theme);
+    enum class Kind {
+        Light,
+        Dark,
+        System,
+    };
+    static Kind kindFromString(QStringView kind);
+
+    static QPalette paletteFromTheme(Kind theme);
```

Comment:
What's your reason for making this an enum? I planned to allow loading other palettes from files in the future, which is why this is currently still a string. Does this have any benefits apart from reducing a few lines of string duplication?

Reply:
I added a commit for a custom _plasma_ style that reads off Plasma APIs directly and bypasses the QQC style. As a pre-work, I made it an enum because I wanted warnings about unimplemented switch cases. Unfortunately the results weren't that much better than using the `System` style. So, I removed the second commit and left the first refactor as-is. 

I did not know about the palettes idea - I think we could still emulate sum types with a `std::variant` compared to using a string. I'll remove this commit if you wish :)

Reply:
Yeah, I would prefer to only include relevant changes for now.